### PR TITLE
Add support for caching dependencies

### PR DESCRIPTION
**Pull request by japarson**
This action has 3 dependencies:
- Trusted Signing PS Module
- Windows SDK Build Tools NuGet Package
- Trusted Signing Client NuGet Package

We can lower execution time by caching these dependencies with GitHub's caching system. While the speedup isn't tremendous on average, it should be a massive boost in the worst-case scenario where the PowerShell Gallery decides to lag which happens quite often.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -235,6 +235,8 @@ timeout: 600
 
 # The summed length of file paths that can be signed with each signtool call. This parameter should only be relevant if you are signing a large number of files. Increasing the value may result in performance gains at the risk of potentially hitting your system's maximum command length limit. The minimum value is 0 and the maximum value is 30000. A value of 0 means that every file will be signed with an individual call to signtool.
 batch-size: 10000
+
+# A boolean value (true/false) that indicates if the dependencies for this action should be cached by GitHub or not. When using self-hosted runners, caches from workflow runs are stored on GitHub-owned cloud storage. A customer-owned storage solution is only available with GitHub Enterprise Server. When enabled, this option can reduce the duration of the action by at least 1 minute. More info: https://docs.github.com/actions/using-workflows/caching-dependencies-to-speed-up-workflows
 ```
```
Comment (dlemstra) [inline README.md:240]: This is missing the `cache-dependencies` name? And it also not clear that this is enabled by default.

File: action.yml [hunk 2]
```diff
@@ -172,10 +172,62 @@ inputs:
...
+        "PSMODULEPATH=$defaultPath" | Out-File -FilePath $env:GITHUB_OUTPUT -Append
+      if: ${{ inputs.cache-dependencies == 'true' }}
+
+    - name: Cache TrustedSigning PowerShell module
+      id: cache-module
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-module
+      with:
+        path: ${{ steps.set-psmodulepath.outputs.PSMODULEPATH }}\TrustedSigning\0.3.18
```
Comment (dlemstra) [inline action.yml:201]: Maybe `0.3.18` and other version should be put in a variable instead?

Comment (japarson) [reply action.yml]: Created a variable via build output in an earlier step.

File: README.md (hunk 1 above)
Comment (japarson) [reply README.md:241]: This is why I shouldn't make PRs at midnight. Thank you for catching that. 😆

**Review by dlemstra — APPROVED** [review]
Thanks for these changes! I have restarted the workflow (https://github.com/ImageMagick/ImageMagick/actions/runs/9257345710) and it is still working.

**Review by chabiss — APPROVED** [review]

**Pull request merged**